### DISCUSSION TITLE
Set theme from HTML

### DISCUSSION THREAD

**Discussion by arthurgeek** [mantinedev/mantine#662] [Q&A] [answered]
Hi 👋,

Thank you for this project. It's really great and the docs are awesome!

 have a question: is it possible to set the theme from HTML, either via class (like Tailwind does) or via some attribute like `data-theme`? I'd like to detect user preference from plain Javascript (via a simple script that runs synchronously before the React or anything, like https://github.com/donavon/use-dark-mode/blob/develop/noflash.js.txt) and set  the theme. This would avoid flashing because we've set the theme before the app loads.

**Comment by rtivital** [comment, marked as answer]
Hi Arthur, thanks for kind words! Mantine does not support getting theme from html, instead you can save theme to local storage or global variable and then simply set it on MantineProvider on first mount.

**Reply by arthurgeek** [reply]
Thanks for the fast reply.

But this won't avoid flash when component is server-side-rendered, right? Because first "mount" happens on the server, there's no localStorage, and it'll flash before `hydrate`.

If I can change HTML and ensure dar theme is applied before the user sees anything, I can avoid flashing.

**Reply by rtivital** [reply]
Currently, setting value from html is not supported, you will need to store data of user preferred color scheme on backend and insert it into MantineProvider during ssr. For example, you can store it in cookie.

**Reply by arthurgeek** [reply]
That's what I'm doing, but in the very first time they access the website and I try to render according to their preferences, a flashing occurs when their preferences and the default from SSR does not match.

Not a big problem, just wanted to know if it could be set from HTML.

Thanks!

**Comment by p0fi** [comment]
Are there any new plans on supporting a simple mechanism to avoid theme flashing?
